www/js/equation.js

```diff
@@ -39,23 +39,45 @@ define([
 				});
 				return equation;
 			}
-			// is there any significance for the following object? else delete
-			this.mapVariableNodeNames = {};
-			// console.log("            parse: ", expr);
 
 			if(params.nameToId){
 				var nodeList = []; //this holds new node ids and variable objects
 				var variableList = []; // this holds the variable list
 				var connections = []; // this holds the input list
+				var dynamicList = []; //this list holds the prior nodes which are to be created further in controller
+
 				array.forEach(expressions, function(expr){
 					variableList = variableList.concat(expr.variables());
+					var currentPriorList = expr.priors();
 					array.forEach(expr.variables(), function(variable){
 						//This is the case where node names have to be converted to ids
 						//This situation arises from equationDoneHandler
 						//In such situation convert name to id
 						var nodeId = subModel.getNodeIDByName(variable);
 						if(nodeId){
 							expr.substitute(variable,nodeId);
+							if(currentPriorList.length>0){
+								currentPriorList.some(function(eachPrior){
+									if(eachPrior === variable){
+										//If the current variable is a part of prior function,
+										// check if the corresponding node has accumulator set (is dynamic)
+										// If not, throw an error indicating the same
+										if(subModel.getVariableType(nodeId) !== "dynamic"){
+											//Also since there is an error with the current node, already created nodes part of the equation should be deleted
+											//nodeList contains the new nodes added to the model which are part of the current equation
+											array.forEach(nodeList,function(delNode){
+												subModel.deleteNode(delNode.id);
```

Comment:
Ohh ok, will make those changes then, this info has to be communicated back to controller and UI node has to be created for those nodes. (Also showing the error message)